rest-api/dao/participant_counts_over_time_service.py

```diff
@@ -0,0 +1,179 @@
+from participant_enums import EnrollmentStatus
+
+class ParticipantCountsOverTimeService(ParticipantSummaryDao):
+
+  def __init__(self):
+    super(ParticipantSummaryDao, self).__init__(ParticipantSummary)
+
+  def get_filtered_results(self, start_date, end_date, filters, stratification='ENROLLMENT_STATUS'):
+    """Queries DB, returns results in format consumed by front-end
+
+    :param start_date: Start date object
+    :param end_date: End date object
+    :param filters: Objects representing filters specified in UI
+    :param stratification: How to stratify (layer) results, as in a stacked bar chart
+    :return: Filtered, stratified results by date
+    """
+
+    filters_sql = self.get_facets_sql(filters)
+
+    if str(stratification) == 'TOTAL':
+      strata = ['TOTAL']
+      sql = """
+        SELECT
+            SUM(ps_sum.cnt * (ps_sum.day <= calendar.day)) registered_count,
+            calendar.day start_date
+        FROM calendar,
+        (SELECT COUNT(*) cnt, DATE(ps.sign_up_time) day
+        FROM participant p
+        LEFT OUTER JOIN participant_summary ps ON p.participant_id = ps.participant_id
+        %(filters)s
+        GROUP BY day) ps_sum
+        WHERE calendar.day >= :start_date
+        AND calendar.day <= :end_date
+        GROUP BY calendar.day
+        ORDER BY calendar.day;
+      """ % {'filters': filters_sql}
+    elif str(stratification) == 'ENROLLMENT_STATUS':
+      strata = [str(val) for val in EnrollmentStatus]
+
+      # Noteworthy comments / documentation from Dan (and lightly adapted)
+      #
+      # This SQL works OK but hardcodes the baseline questionnaires and the
+      # fact that 1ED04 and 1SAL are the samples needed to be a full
+      # participant into the SQL. Previously this was done with config:
+      #
+      # rest-api/config/base_config.json#L29
+      #
+      # For the samples, MySQL doesn't make it easy to do the LEAST of N
+      # nullable values, so this is probably OK...
+      #
+      # For the baseline questionnaires, note that we might want to use the config to
+      # generate the GREATEST statement instead, but for now are hardcoding as
+      # we do with samples.
+
+      # TODO when implementing unit testing:
+      # Add macros for GREATEST and LEAST, as they don't work in SQLite
+      # Example: master/rest-api/dao/database_utils.py#L50
+
+      sql = """
+      SELECT
+         SUM(registered_cnt * (cnt_day <= calendar.day)) registered_participants,
+         SUM(member_cnt * (cnt_day <= calendar.day)) member_participants,
+         SUM(full_cnt * (cnt_day <= calendar.day)) full_participants,
+         calendar.day
+       FROM
+       (SELECT c2.day cnt_day,
+               registered.cnt registered_cnt,
+               member.cnt member_cnt,
+               full.cnt full_cnt
+            FROM calendar c2
+            LEFT OUTER JOIN
+            (SELECT COUNT(*) cnt, DATE(ps.consent_for_study_enrollment_time) day
```

Comment:
I think you want to change this to p.sign_up_time... otherwise you won't be capturing participants that register but don't consent in the registered numbers.

If you do this, the left outer join below is technically unnecessary... you're not pulling anything from participant summary.

Reply:
I changed it to p.sign_up_time; thanks for the catch.  But I kept the left outer join, as some filters use participant_summary.